base/summarysize.jl

```diff
@@ -139,17 +146,17 @@ end
 function (ss::SummarySize)(obj::Core.TypeName)
     key = pointer_from_objref(obj)
     haskey(ss.seen, key) ? (return 0) : (ss.seen[key] = true)
-    return Core.sizeof(obj)
+    return (ss.count ? 1 : Core.sizeof(obj))
 end
 
 function (ss::SummarySize)(obj::GenericMemory)
     haskey(ss.seen, obj) ? (return 0) : (ss.seen[obj] = true)
-    headersize = 2*sizeof(Int)
-    size::Int = headersize
-    datakey = unsafe_convert(Ptr{Cvoid}, obj)
+    headersize = 2 * sizeof(Int)
+    size::Int = (ss.count ? 1 : headersize)
```

Comment:
@oscardssmith Is the header here separate from the `obj` itself that we count below? I don't quite grok the accounting

Reply:
Thanks! Think this is good now